### PR TITLE
Development.

### PR DESCRIPTION
**Pull request by pashabetenya**
Implementation Pages: Sign Up, Sign In and Recovery. Service Setup.

### CODE REVIEW

**Comment by pashabetenya** [comment]
I do the division into widgets, 
and use a made-up part of the structure. 

Eventually the application will be moved to a clean architecture, 


and comments are for me :)

File: lib/ui/pages/authorization/authorization_page.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import 'components/authorization_description.dart';
+import 'components/authorization_text_button.dart';
+import 'components/authorization_material_button.dart';
+
+class AuthorizationPage extends StatelessWidget {
+  const AuthorizationPage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final height = MediaQuery.of(context).size.height;
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/authorization_page.dart:13]: please do not use mediaQuery. Try to use Expanded widget instead

File: lib/ui/pages/authorization/authorization_page.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            child: panel(height),
+          ),
+        ],
+      ),
+    );
+  }
+
+  // The widget, that draws the content.
+
+  Widget content(height) {
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/authorization_page.dart:49]: please do not use methods, returning widget. Use private widgets instead. Methods broke application performance

File: lib/ui/pages/authorization/authorization_page.dart [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    return Center(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          // Call the widget, that draws the picture + text.
+
+          LogotypeWidget(color: const Color(0xFFF1F1F5), height: height * 0.06),
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/authorization_page.dart:56]: Try to not use height. Use Align, Expanded, Flexible widgets instead

File: lib/ui/pages/authorization/authorization_page.dart [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+          const AuthorizationDescription(),
+        ],
+      ),
+    );
+  }
+
+  // The widget, that draws the panel.
+
+  Widget panel(height) {
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/authorization_page.dart:70]: Do not use methods, returning widgets (Check everywhere)

File: lib/ui/pages/authorization/authorization_page.dart [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    return Align(
+      alignment: Alignment.bottomCenter,
+      child: Column(
+        children: [
+          // Call the widget, that draws the material button.
+
+          const AuthorizationMaterialButton(),
+
+          SizedBox(height: height * 0.02),
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/authorization_page.dart:79]: Do not use height. (Check everywhere)

File: lib/ui/pages/authorization/components/authorization_description.dart [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    final sFactor = MediaQuery.of(context).textScaleFactor;
+
+    return Text(
+      LocaleKeys.authorization_description.tr(),
+      textAlign: TextAlign.center,
+      style: TextStyle(
+        fontFamily: 'MontserratRegular',
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/components/authorization_description.dart:16]: Too many parameters of TextStyle. Maybe better to create TextStyle presets and use them across the app?

File: lib/ui/pages/authorization/components/authorization_material_button.dart [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import 'package:application/ui/pages/registry/registry_page.dart';
+import 'package:application/generated/locale_keys.g.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:application/ui/widgets/button/material_button_widget.dart';
+
+class AuthorizationMaterialButton extends StatelessWidget {
+  const AuthorizationMaterialButton({Key? key}) : super(key: key);
+
+  void transition(context) {
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/components/authorization_material_button.dart:10]: bad method name.
Also, method should be private. 
Place private methods below the public methods

File: lib/ui/pages/authorization/components/authorization_material_button.dart [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final font = MediaQuery.of(context).size.width;
+
+    return MaterialButtonWidget(
+      color: const Color(0xFF12CDD9),
+      height: MediaQuery.of(context).size.height * 0.054,
+      width: MediaQuery.of(context).size.width,
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/components/authorization_material_button.dart:24]: Again strange calculations. Rid of it

File: lib/ui/pages/authorization/components/authorization_text_button.dart [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    );
+  }
+}
+
+class AuthorizationTextButtonStyle {
+  AuthorizationTextButtonStyle(this.font);
+
+  final double font;
+
+  TextStyle get textStyle1 => TextStyle(
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/components/authorization_text_button.dart:67]: textStyle1? What does it means?

File: lib/ui/pages/authorization/components/authorization_text_button.dart [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        fontFamily: 'MontserratRegular',
+        fontSize: font,
+        fontWeight: FontWeight.w500,
+        height: 17.6 / font,
+        letterSpacing: 0.12,
+        color: const Color(0xFF92929D),
+      );
+
+  TextStyle get textStyle2 => TextStyle(
```
Comment (Vovanella95) [inline lib/ui/pages/authorization/components/authorization_text_button.dart:76]: Same comment as for textStyle1

File: lib/ui/pages/login/components/login_text_button.dart [hunk 11]
```diff
@@ -0,0 +1,43 @@
+import 'package:application/generated/locale_keys.g.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:application/ui/pages/recovery/recovery_page.dart';
+
+class LoginTextButton extends StatelessWidget {
+  const LoginTextButton({Key? key}) : super(key: key);
+
+  void transition(context) {
```
Comment (Vovanella95) [inline lib/ui/pages/login/components/login_text_button.dart:9]: Bad method name

File: lib/ui/pages/login/components/login_text_button.dart [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    Navigator.push(
+      context,
+      MaterialPageRoute(builder: (context) => const RecoveryPage()),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final width = MediaQuery.of(context).size.width;
```
Comment (Vovanella95) [inline lib/ui/pages/login/components/login_text_button.dart:18]: Rid of width and calculations

File: lib/ui/pages/login/login_page.dart [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+              children: [
+                // Call the widget, that draws the description text.
+                // + title text.
+
+                AuthorizationDescriptionWidget(
+                  title: LocaleKeys.login_title.tr(),
+                  description: LocaleKeys.login_description.tr(),
+                ),
+
+                SizedBox(height: height * 0.085),
```
Comment (Vovanella95) [inline lib/ui/pages/login/login_page.dart:54]: Remove dependence from height

File: lib/ui/pages/onboarding/components/onboarding_indicator.dart [hunk 14]
```diff
@@ -1,10 +1,7 @@
 import 'package:flutter/material.dart';
 
 class OnboardingIndicator extends StatelessWidget {
-  const OnboardingIndicator({
-    Key? key,
-    required this.indicator,
-  }) : super(key: key);
+  const OnboardingIndicator({Key? key, required this.indicator}) : super(key: key);
```
Comment (Vovanella95) [inline lib/ui/pages/onboarding/components/onboarding_indicator.dart:4]: required properties should be placed before optional

File: lib/ui/pages/onboarding/components/onboarding_material_button.dart [hunk 15]
```diff
@@ -0,0 +1,33 @@
+import 'package:application/ui/widgets/button/material_button_widget.dart';
+import 'package:application/ui/pages/authorization/authorization_page.dart';
+import 'package:flutter/material.dart';
+
+class OnboardingMaterialButton extends StatelessWidget {
+  const OnboardingMaterialButton({Key? key}) : super(key: key);
+
+  void transition(context) {
```
Comment (Vovanella95) [inline lib/ui/pages/onboarding/components/onboarding_material_button.dart:8]: Bad method name

**Review by Vovanella95 — CHANGES_REQUESTED** [review]
Please, fix comments.

**Pull request merged**